File: TVTComment/Model/IPC/IPCMessage/CommandIPCMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TVTComment.Model.IPC.IPCMessage
{
    class CommandIPCMessage:IIPCMessage
    {
        public string MessageName=>"Command";

        public string CommandId { get; private set; }

        public IEnumerable<string> Encode()
        {
            return new string[1] { CommandId };
        }

        public void Decode(IEnumerable<string> content)
        {
            string[] contents = content.ToArray();
            if (contents.Length != 1)
                throw new IPCMessageDecodeException("CurrentChannelのcontentの数が1以外です");

            CommandId = contents[0];
        }
    }
}

File: Viewer/TVTComment/IPC/IPCMessage/CommandIPCMessage.cpp
#include "stdafx.h"
#include "CommandIPCMessage.h"
#include "../IPCMessageDecodeError.h"

namespace TVTComment
{
	std::string CommandIPCMessage::GetMessageName() const
	{
		return "Command";
	}

	std::vector<std::string> CommandIPCMessage::Encode() const
	{
		return{this->CommandId};
	}

	void CommandIPCMessage::Decode(const std::vector<std::string> &contents)
	{
		if (contents.size() != 1)
			throw IPCMessageDecodeError("CommandIPCMessageのcontentsの数が1以外です");
		
		this->CommandId = contents[0];
	}
}